cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java

```diff
@@ -930,8 +953,9 @@ public void getProceduresByApp(HttpRequest request, HttpResponder responder,
   @Path("/apps/{app-id}/mapreduce")
   public void getMapreduceByApp(HttpRequest request, HttpResponder responder,
                                 @PathParam("app-id") String appId) {
-    programLifecycleHttpHandler.getProgramsByApp(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, appId,
-                                                 ProgramType.MAPREDUCE.getCategoryName());
+    programLifecycleHttpHandler.getProgramsByApp(responder,
```

Comment:
In all these methods, better to call the corresponding method in the v3 handler instead of the generic ``getProgramsByApp``. Then make ``getProgramsByApp`` private in ``ProgramLifecycleHttpHandler``?